Added an AppSettings option to disable Sqlite connection pooling.

johnrom opened the pull request:
closes #11387 

Sqlite locks the file when connection pooling is enabled, which can prevent certain backup processes, App Service cloning processes, etc from working. This PR adds an option to disable connection pooling and remove the lock from the Sqlite DB when it isn't being written to.

### Testing

- Pull this PR
- Launch .Net Core Launch (web debug)
- Install the site with an Sqlite Multitenant profile.
- Try to delete `App_Data/Sites/Default/yessql.db`
  > ❌ You'll get an error because the file is locked.
- Stop the application
- Open `OrchardCore.Cms.Web/appsettings.json` and uncomment these lines:
  ```json
  "OrchardCore_Data_Sqlite": {
    "UseConnectionPooling": false
  },
  ```
- Launch .Net again
- Delete `App_Data/Sites/Default/yessql.db`
  >  ✓ The file will be deleted.

Skrypt commented:
Why not. The code is great.

hishamco reviewed (changes requested):
LGTM, it would be nice if you added docs into the introduced types

ns8482e commented:
@johnrom @Skrypt lets also add database file name as part of options

hishamco commented:
Good point Niraj I already created an old PR, I think it's the time to make configurable

ns8482e commented:
These properties are also worth (debatable) adding to  `SqliteOptions`

- DatabaseFileName ( with default value to `yessql.db`)
- Max Pool Size
- DateTimeFormat 
- Cached
- CacheSize
- Password

hishamco commented:
> DatabaseFileName ( with default value to yessql.db)

`OC.db` ;)

> Password (default empty)

I'm not sure if it's really works, I try it long time back with no luck

sebastienros commented:
At this point with all the options it's just a connection string, so might as well allow for custom connection strings on setup that can also be changed in the settings (like the other databases). And probably provide a default value for sqlite.

sebastienros commented:
But here it's a global app option to disable pooling, so I'd rather use this PR as-is without other "tenant-specific" options like filename and so on.

johnrom commented:
@hishamco I added docs to the docs, but neglected the types. I'll try and circle back today or tomorrow if the PR doesn't get shut down in favor of a raw connection string.

@ns8482e I wanted to keep this PR specific to the issue I was trying to solve -- it can serve as a basis for additional configuration in a separate issue / PR.

@sebastienros It could probably be a connection string, but I was considering the fact that OC might change how it works depending on whether certain options are default or not. I personally would rather provide specific options and then build the connection string using ConnectionStringBuilder instead of accepting any old value because:

- Removing Pooling or changing Caching behavior etc, might someday require some other service to act differently.
- A new feature might be released for Sqlite and someone might opt in even though OC would completely break because of it.

ns8482e commented:
> @ns8482e I wanted to keep this PR specific to the issue I was trying to solve -- it can serve as a basis for additional configuration in a separate issue / PR.

IMHO It's worth adding more options to `SqliteOptions`,  and I also agree it shouldn't be  blocking this PR

ns8482e commented on src/OrchardCore/OrchardCore.Data.YesSql/Options/SqliteOptions.cs:
```diff
@@ -0,0 +1,7 @@
+{
+    public class SqliteOptions
+    {
+        public bool PoolConnections { get; set; } = true;
```
Nice if we use Has/Is/Can prefix for `bool`

hishamco replied:
What about `UsePooling`?

ns8482e replied:
I am fine - it just `PoolConnections` feels like #s  (list/array kind)

johnrom replied:
Changed to `UseConnectionPooling`.

johnrom commented:
Switched option name to `UseConnectionPooling` per https://github.com/OrchardCMS/OrchardCore/pull/11393#pullrequestreview-913072831 and added doc comments where necessary per https://github.com/OrchardCMS/OrchardCore/pull/11393#pullrequestreview-912928472.
